precompiles/proxy/src/lib.rs

```diff
@@ -193,4 +269,83 @@ where
 
 		Ok(is_proxy)
 	}
+
+	fn inner_proxy(
+		handle: &mut impl PrecompileHandle,
+		real: Address,
+		force_proxy_type: Option<<Runtime as pallet_proxy::Config>::ProxyType>,
+		evm_subcall: EvmSubCall,
+	) -> EvmResult {
+		// Proxied call can be dispatch by users only.
+		// We should forbid precompiles here because pre_check allow precompiles.
+		if <Runtime as pallet_evm::Config>::PrecompilesValue::get()
+			.is_precompile(handle.context().caller)
+		{
+			return Err(revert("Proxy.proxy not callable by precompiles"));
```

Comment:
What happens with this check if a SC uses Proxy to call the Randomness precompile, which then calls the fulfillment back? :thinking:

Reply:
>  if a SC uses Proxy

A SC can't use proxy precompile. The `pre_check` function not allow callers with code.